### PR TITLE
Skip deeply-recursive test in `System.Runtime.Serialization.Xml.ReflectionOnly.Tests` if doing random OSR

### LINKED ISSUE #100246: Test `System.Runtime.Serialization.Xml.ReflectionOnly.Tests` stack overflowing in Windows x64 release for `libraries-pgo` pipeline
Some tests in `System.Runtime.Serialization.Xml.ReflectionOnly.Tests`  are failing due to a stack overflow on windows x64 for `libraries-pgo` pipeline.

[Link
](https://dev.azure.com/dnceng-public/public/_build/results?buildId=616074&view=logs&j=6f1019c2-8724-53e2-23d1-81bb32b7e0cf&t=e0493a78-ae3a-5d09-cb3c-5f65905d16cd)
```
set DOTNET_TieredCompilation=1
set DOTNET_TC_OnStackReplacement=1
set DOTNET_TC_QuickJitForLoops=1
set DOTNET_TC_OnStackReplacement_InitialCounter=1
set DOTNET_OSR_HitLimit=2
set DOTNET_JitRandomOnStackReplacement=15

C:\h\w\A5B1092D\w\B8060970\e>"C:\h\w\A5B1092D\p\dotnet.exe" exec --runtimeconfig System.Runtime.Serialization.Xml.ReflectionOnly.Tests.runtimeconfig.json --depsfile System.Runtime.Serialization.Xml.ReflectionOnly.Tests.deps.json xunit.console.dll System.Runtime.Serialization.Xml.ReflectionOnly.Tests.dll -xml testResults.xml -nologo -nocolor -notrait category=IgnoreForCI -notrait category=OuterLoop -notrait category=failing  
  Discovering: System.Runtime.Serialization.Xml.ReflectionOnly.Tests (method display = ClassAndMethod, method display options = None)
  Discovered:  System.Runtime.Serialization.Xml.ReflectionOnly.Tests (found 260 of 261 test cases)
  Starting:    System.Runtime.Serialization.Xml.ReflectionOnly.Tests (parallel test collections = on [4 threads], stop on fail = off)
Stack overflow.
   at System.Xml.ValueHandle.GetCharsText()
   at System.Xml.ValueHandle.GetString()
```

### PR DESCRIPTION
**Pull request by amanasifkhalid**
Fixes #100246. Disables test that frequently overflows the stack for OSR stress scenarios (see [comment](https://github.com/dotnet/runtime/issues/100246#issuecomment-2278801505) for discussion).

### CODE REVIEW

File: src/libraries/System.Runtime.Serialization.Xml/tests/DataContractSerializer.cs [hunk 1]
```diff
@@ -4477,8 +4477,11 @@ public static void DCS_TypeWithPrimitiveKnownTypes()
         Assert.NotNull(actual);
     }
 
+    // Random OSR might cause a stack overflow on Windows x64
+    private static bool IsNotWindowsRandomOSR => !PlatformDetection.IsWindows || (Environment.GetEnvironmentVariable("DOTNET_JitRandomOnStackReplacement") == null);
+
     [SkipOnPlatform(TestPlatforms.Browser, "Causes a stack overflow")]
```
Comment (amanasifkhalid) [inline src/libraries/System.Runtime.Serialization.Xml/tests/DataContractSerializer.cs:4483]: I probably should've noticed this hint earlier...

**Comment by amanasifkhalid** [comment]
@AndyAyersMS PTAL, the last `libraries-pgo` [run](https://dev.azure.com/dnceng-public/public/_build/results?buildId=774538&view=results) didn't hit the stack overflow.

**Review by AndyAyersMS — APPROVED** [review]
LGTM

**Pull request merged**